crates/iceberg-catalog/src/service/storage/gcs/sts.rs

```diff
@@ -0,0 +1,226 @@
+                    available_permissions: vec![
+                        match storage_permissions {
+                            StoragePermissions::Read => "inRole:roles/storage.objectViewer".to_string(),
+                            StoragePermissions::ReadWrite => "inRole:roles/storage.objectCreator".to_string(),
+                            StoragePermissions::ReadWriteDelete => "inRole:roles/storage.objectUser".to_string()
+                        },
+                    ],
+                    availability_condition: AvailabilityCondition {
+                        title: "obj-prefixes".to_string(),
+                        // need the getAttribute to allow Listing operations
+                        expression: format!(
+                            "resource.name.startsWith('projects/_/buckets/{bucket}/objects/{prefixless_location}') || api.getAttribute('storage.googleapis.com/objectListPrefix', '').startsWith('{prefixless_location}')",
+                        ),
+                    },
+                }],
+            },
+
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct AccessBoundary {
+    #[serde(rename = "accessBoundaryRules")]
+    access_boundary_rules: Vec<AccessBoundaryRule>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct AccessBoundaryRule {
+    #[serde(rename = "availableResource")]
+    available_resource: String,
+    #[serde(rename = "availablePermissions")]
+    available_permissions: Vec<String>,
+    availability_condition: AvailabilityCondition,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
+struct AvailabilityCondition {
+    title: String,
+    expression: String,
+}
+
+impl From<&GcsServiceKey> for CredentialsFile {
+    fn from(
+        GcsServiceKey {
+            r#type: tp,
+            project_id,
+            private_key_id,
+            private_key,
+            client_email,
+            client_id,
+            auth_uri,
+            token_uri,
+            auth_provider_x509_cert_url: _,
+            client_x509_cert_url: _,
+            universe_domain: _,
+        }: &GcsServiceKey,
+    ) -> Self {
+        Self {
+            tp: tp.to_string(),
+            client_email: Some(client_email.clone()),
+            private_key_id: Some(private_key_id.clone()),
+            private_key: Some(private_key.clone()),
+            auth_uri: Some(auth_uri.clone()),
+            token_uri: Some(token_uri.clone()),
+            project_id: Some(project_id.clone()),
+            client_secret: None,
+            client_id: Some(client_id.clone()),
+            refresh_token: None,
+            audience: None,
+            subject_token_type: None,
+            token_url_external: None,
+            token_info_url: None,
+            service_account_impersonation_url: None,
+            service_account_impersonation: None,
+            delegates: None,
+            credential_source: None,
+            quota_project_id: None,
+            workforce_pool_user_project: None,
+        }
+    }
+}
```

Comment:
I think we should add a test here that tests the negative case - i.e. that a file outside of the downscoped prefix cannot be accessed. The validate method only tests the positive case I think.

Reply:
added tests for vended creds in `service/storage/mod.rs`